### PR TITLE
Implement update notifications

### LINKED ISSUE #497: Update Notifications
Almost done, just need to do the JSON parsing

### PR DESCRIPTION
**Pull request by CaptainElliott**
This file currently needs to be created https://github.com/CaptainElliott/CardinalPGM/commit/ff0362058597321c0c614f7a1d6b097b4fb590d1#diff-f0e9d0bcff46f7f9cccd4180b4610642R18

following the same schema as https://raw.githubusercontent.com/CaptainElliott/test-ignore/master/update.json

It works by checking if the git version of the plugin is the same as the one in this repository, if not it gets the information from that file

closes #497

### CODE REVIEW

File: src/main/java/in/twizmwaz/cardinal/util/GitUtils.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.google.gson.JsonParser;
+
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+public class GitUtils {
+
+    public static String getLatestGitRevision(){
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/util/GitUtils.java]: Missing a space

**Comment by twizmwazin** [comment]
- The update message should just be a Minecraft Json, which can be sent directly to the player. This would allow the "update" notification to be used outside of a specific format. We don't need the other fields, they can be included in the message field.
- The git revision should be stored as-is in the plugin, storing it just as a plain UTF-8 file in the jar would be sufficient. I don't like the idea of extracting it from the version string.

**Comment by CaptainElliott** [comment]
Ah, I get what you mean by minecraft json now :p

**Comment by twizmwazin** [comment]
@CaptainElliott Just make a file with ${describe} in the resources folder, it will get filtered by maven.

**Comment by CaptainElliott** [comment]
@twizmwazin could you create the repo with something like this inside 
https://raw.githubusercontent.com/CaptainElliott/test-ignore/master/update.json

**Comment by twizmwazin** [comment]
@CaptainElliott I'll do it as soon as I get home, at school. I'll be home approx 4:15 my time.

**Comment by ryanw-ny** [comment]
@CaptainElliott I would wrap the SportBukkit server itself with a launch application, that can handle doing the updates, and other things. You could also have it automatically download SportBukkit, and have complete control over using the latest versions of things, etc.

**Comment by CaptainElliott** [comment]
@ryanw-ny The Git API is being used and we have talked about having cardinal auto update itself in the future, not sure about SportBukit, but I don't think that needs to be done.

There's no harm in setting it to false :package: 

@twizmwazin ${describe} will return the whole version including artifact ID and only the short version of the commit, which is how I did it originally

**Comment by CaptainElliott** [comment]
https://raw.githubusercontent.com/CaptainElliott/test-ignore/master/update.json Looks like

File: src/main/java/in/twizmwaz/cardinal/UpdateHandler.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        this.message = GitUtils.getUpdateMessage(updateLocation);
+        this.localGitRevision = GameHandler.getGameHandler().getPlugin().getDescription().getVersion().substring(GameHandler.getGameHandler().getPlugin().getDescription().getVersion().length() - 6, GameHandler.getGameHandler().getPlugin().getDescription().getVersion().length());
+    }
+
+    public void sendUpdateMessage(Player player) {
+        IChatBaseComponent chat = ChatSerializer.a(message);
+        PacketPlayOutChat packet = new PacketPlayOutChat(chat);
+
+        EntityPlayer nmsPlayer = ((CraftPlayer) player).getHandle();
+        nmsPlayer.playerConnection.sendPacket(packet);
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/UpdateHandler.java]: The NMS is unnecessary, use player#sendRawMessage(String json)

File: src/main/java/in/twizmwaz/cardinal/module/modules/updateNotification/UpdateNotification.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    @Override
+    public void unload() {
+        HandlerList.unregisterAll(this);
+    }
+
+    @EventHandler
+    public void onPlayerJoin(PlayerJoinEvent event) {
+        if (event.getPlayer().isOp()) {
+            try {
+                updateHandler = new UpdateHandler();
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/module/modules/updateNotification/UpdateNotification.java]: Why can't we just make one instance and use that?

File: src/main/java/in/twizmwaz/cardinal/UpdateHandler.java (hunk 2 above)
Comment (CaptainElliott) [inline src/main/java/in/twizmwaz/cardinal/UpdateHandler.java]: sendRawMessage is not the same as /tellraw

Comment (dentmaged) [inline src/main/java/in/twizmwaz/cardinal/UpdateHandler.java]: Can't you deserialise it then use player#sendMessage(String message)?

**Comment by twizmwazin** [comment]
Commits pushed manually

**Pull request closed without merging**